src/components/Table/index.tsx

```diff
@@ -1,28 +1,42 @@
 import React, { ReactNode, Component, MouseEvent } from 'react';
 import { DragDropContext, Droppable, DropResult } from 'react-beautiful-dnd';
 import StyledTable from './style';
+import Card from './Card';
 import Row from './Row';
+import TableHeaders from './TableHeaders';
+import CompactHeaders from './CompactHeaders';
 import Branch from '../Branch';
-import Headers from './Headers';
+import { MediumIcons } from './../Icon/types';
+import SeverityType from '../../types/_SeverityType';
 
 type SortDirectionType = 'ascending' | 'descending' | 'none';
 
 type BaseRowType = {
     id: string;
     selected?: boolean;
+    buttons?: Array<ReactNode>;
+    statusIcons?: Array<StatusIconType>;
     // tslint:disable-next-line
-    [key: string]: string | number | boolean | undefined;
+    [key: string]: string | number | boolean | undefined | Array<ReactNode> | ReactNode;
+};
+
+type StatusIconType = {
+    label: string;
+    severity?: SeverityType;
+    icon: keyof typeof MediumIcons;
 };
 
 type ColumnType<GenericCellType, GenericRowType> = {
     order?: number;
-    header?: ReactNode;
+    header?: ReactNode | string | number | any;
     align?: 'start' | 'center' | 'end';
+    width?: string;
     sort?(cellA: GenericCellType, cellB: GenericCellType): number;
     render?(cell: GenericCellType, row: GenericRowType): JSX.Element;
 };
 
 type PropsType<GenericRowType extends BaseRowType> = {
+    view?: 'table' | 'datacard';
```

Comment:
Styled components added an `as` prop for this kind of behavior in v4 (which was updated in #227) maybe it could be nice to as that convention here as well, since we already do it for Text/Heading etc. Also, maybe "card" is a better wording here since data is a but implied when rendering a table :)

```suggestion
   as?: 'table' | 'card';
```

Reply:
Okay, good suggestion. I changed the `view` prop to `as` everywhere.